perfkitbenchmarker/providers/openstack/os_disk.py

```diff
@@ -59,34 +61,32 @@ def _Delete(self):
             self.__nclient.volumes.delete(self._disk)
             is_deleted = False
             while not is_deleted:
-                is_deleted = len(self.__nclient.volumes.findall(
-                    display_name=self.name)) == 0
+                volume = self.__nclient.volumes.get(self._disk.id)
+                is_deleted = volume is None
                 time.sleep(sleep)
                 sleep_count += 1
                 if sleep_count == 10:
                     sleep = 5
-        except (os_utils.NotFound, os_utils.BadRequest):
+        except Exception:
```

Comment:
Would it be worth including some information on this in the logging statement? This could swallow an exception unrelated to the volume not being found...